admin/src/containers/MainLayout/PanelContext.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component, createContext } from 'react';
+import PropTypes from 'prop-types';
+
+const PanelContext = createContext({
+  component: null,
+  title: null,
+  props: {},
+  openPanel: () => {},
+  closePanel: () => {},
+  isPanelOpen: false
+});
+
+export class PanelProvider extends Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.element,
+      PropTypes.object,
+      PropTypes.arrayOf(PropTypes.element)
+    ])
+  };
+
+  openPanel = (component, title = null, props = {}) => {
+    this.setState({
+      component,
+      title,
+      isPanelOpen: true,
+      props
+    });
+  };
+
+  closePanel = () =>
```

Comment:
Same than above!

Reply:
This syntax ensures `this` is bound within the fn() but it's also experimental. It's supposed to be safe with babel, but I use it on prototypes :P It can be safer to fn() {} and bind `this` in the constructor() :D

Reply:
Oh, now I understand. We can experiment without assigning the function to a property. I'll play with this in a next iteration.